PennMobile/src/main/java/com/pennapps/labs/pennmobile/GsrFragment.kt

```diff
@@ -0,0 +1,519 @@
+
+    // api manager
+    private var mLabs: Labs? = null
+
+    //list that holds all GSR rooms
+    private val gsrHashMap = HashMap<String, Int>()
+
+    private var gsrRoomListRecylerView: RecyclerView? = null
+
+    private var gsrLocationsArray = ArrayList<String>()
+
+    // all the gsrs
+    private var mGSRS = ArrayList<GSRContainer>()
+
+    private var formatter = DateTimeFormat.forPattern("yyyy-MM-dd'T'HH:mm:ss")
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        mLabs = MainActivity.getLabsInstance()
+        (activity as MainActivity).closeKeyboard()
+        activity?.let {activity ->
+            activity.setTitle(R.string.gsr)
+        }
+        // fabric report handling
+        Fabric.with(context, Crashlytics())
+        Answers.getInstance().logContentView(ContentViewEvent()
+                .putContentName("GSR")
+                .putContentType("App Feature")
+                .putContentId("0"))
+    }
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+
+        // link UI elements
+        val v = inflater.inflate(R.layout.fragment_gsr, container, false)
+        calendarButton = v.select_date
+        startButton = v.select_start_time
+        endButton = v.select_end_time
+        gsrDropDown = v.gsr_building_selection
+        instructions = v.instructions
+
+        // populate the list of gsrs
+        populateDropDownGSR()
+
+        // Get calendar time and date: standardize to EST
+        val tz = TimeZone.getTimeZone("America/New_York")
+        val calendar = Calendar.getInstance(tz)
+        val minutes = calendar.get(Calendar.MINUTE)
+        val hour = calendar.get(Calendar.HOUR)
+        val day = calendar.get(Calendar.DAY_OF_MONTH)
+        val month = calendar.get(Calendar.MONTH) + 1
+        val year = calendar.get(Calendar.YEAR)
+        val ampm = calendar.get(Calendar.AM_PM)
+
+        calendarButton?.text = (month.toString() + "/" + day + "/" + year)
+
+        // Set default start/end times for GSR booking
+        val ampmTimes = getStartEndTimes(hour, minutes, ampm)
+        startButton?.text = ampmTimes[0]
+        endButton?.text = ampmTimes[1]
+
+        // Set up recycler view for list of GSR rooms
+        gsrRoomListRecylerView = v.findViewById<View>(R.id.gsr_rooms_list) as RecyclerView
+
+        /**
+         * On Click functions for buttons
+         */
+
+        //set start time button
+        startButton?.setOnClickListener(View.OnClickListener {
+            // Get Current Time
+            val c = Calendar.getInstance()
+            val mHour = c.get(Calendar.HOUR_OF_DAY)
+            val mMinute = c.get(Calendar.MINUTE)
+
+            // Launch Time Picker Dialog
+            val timePickerDialog = TimePickerDialog(activity,
+                    TimePickerDialog.OnTimeSetListener { view, hourOfDay, minute ->
+                        val AM_PM: String
```

Comment:
You could use a ternary operator here:

`val AM_PM = if (hourOfDay < 12) "AM" else "PM"`